pkg/mimir/mimir.go

```diff
@@ -237,6 +237,9 @@ func (c *Config) Validate(log log.Logger) error {
 	if err := c.IngesterClient.Validate(log); err != nil {
 		return errors.Wrap(err, "invalid ingester_client config")
 	}
+	if err := c.Ingester.IngesterRing.Validate(log); err != nil {
```

Comment:
From a design perspective, the `Validate()` is meant to be cascading. So you should add `Validate()` to `ingester.Config`. `ingester.Config.Validate()` should call `IngesterRing.Validate()`.